code/game/objects/items/choice_beacon.dm

```diff
@@ -6,140 +6,170 @@
 	inhand_icon_state = "radio"
 	lefthand_file = 'icons/mob/inhands/items/devices_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/items/devices_righthand.dmi'
+	/// How many uses this item has before being deleted
 	var/uses = 1
+	/// Used in the deployment message - What company is sending the equipment, flavor
+	var/company_source = "Central Command"
+	/// Used inthe deployment message - What is the company saying with their message, flavor
+	var/company_message = span_bold("Item request received. Your package is inbound, please stand back from the landing site.")
+
+/obj/item/choice_beacon/interact(mob/user)
+	. = ..()
+	if(!can_use_beacon(user))
+		return
 
-/obj/item/choice_beacon/attack_self(mob/user)
-	if(canUseBeacon(user))
-		generate_options(user)
+	open_options_menu(user)
 
-/obj/item/choice_beacon/proc/generate_display_names() // return the list that will be used in the choice selection. entries should be in (type.name = type) fashion. see choice_beacon/hero for how this is done.
+/// Return the list that will be used in the choice selection.
+/// Entries should be in (type.name = type) fashion.
+/obj/item/choice_beacon/proc/generate_display_names()
 	return list()
 
-/obj/item/choice_beacon/proc/canUseBeacon(mob/living/user)
+/// Checks if this mob can use the beacon, returns TRUE if so or FALSE otherwise.
+/obj/item/choice_beacon/proc/can_use_beacon(mob/living/user)
 	if(user.canUseTopic(src, be_close = TRUE, no_dexterity = FALSE, no_tk = TRUE))
 		return TRUE
-	else
-		playsound(src, 'sound/machines/buzz-sigh.ogg', 40, TRUE)
-		return FALSE
 
-/obj/item/choice_beacon/proc/generate_options(mob/living/M)
+	playsound(src, 'sound/machines/buzz-sigh.ogg', 40, TRUE)
+	return FALSE
+
+/// Opens a menu and allows the mob to pick an option from the list
+/obj/item/choice_beacon/proc/open_options_menu(mob/living/user)
 	var/list/display_names = generate_display_names()
 	if(!length(display_names))
 		return
-	var/choice = tgui_input_list(M, "Which item would you like to order?", "Select an Item", display_names)
-	if(isnull(choice))
-		return
-	if(isnull(display_names[choice]))
+	var/choice = tgui_input_list(user, "Which item would you like to order?", "Select an Item", display_names)
+	if(isnull(choice) || isnull(display_names[choice]))
 		return
-	if(!M.canUseTopic(src, be_close = TRUE, no_dexterity = FALSE, no_tk = TRUE))
+	if(!can_use_beacon(user))
 		return
 
-	spawn_option(display_names[choice],M)
-	uses--
-	if(!uses)
+	consume_use(display_names[choice], user)
+
+/// Consumes a use of the beacon, sending the user a message and creating their item in the process
+/obj/item/choice_beacon/proc/consume_use(obj/choice_path, mob/living/user)
+	to_chat(user, span_hear("You hear something crackle from the beacon for a moment before a voice speaks. \
+		\"Please stand by for a message from [company_source]. Message as follows: [company_message] Message ends.\""))
+
+	spawn_option(choice_path, user)
+	if(--uses <= 0)
```

Comment:
I hate this

Reply:
@Time-Green @MrMelbert You are not supposed to do this because it sucks and is hard to read

Reply:
Sorry, I try to keep personel preference out of it, but I guess it wasn’t just me

Reply:
ur job is to enforce your personal preferences to an extent